proposals/dao/fip_92.ts

```diff
@@ -0,0 +1,177 @@
+    false, // bool _feiIsReportedIn
+    false // bool _feiIsOther
+  );
+  await balancerLensVeBalBal.deployTransaction.wait();
+  logging && console.log('balancerLensVeBalBal: ', balancerLensVeBalBal.address);
+
+  // Deploy lens to report the WETH part of B-80BAL-20WETH vote-locked
+  const balancerLensVeBalWeth = await balancerPool2LensFactory.deploy(
+    veBalDelegatorPCVDeposit.address, // address _depositAddress
+    addresses.wethERC20, // address _token
+    '0x5c6ee304399dbdb9c8ef030ab642b10820db8f56', // IWeightedPool _pool
+    addresses.chainlinkEthUsdOracleWrapper, // IOracle _reportedOracle
+    addresses.balUsdCompositeOracle, // IOracle _otherOracle
+    false, // bool _feiIsReportedIn
+    false // bool _feiIsOther
+  );
+  await balancerLensVeBalWeth.deployTransaction.wait();
+  logging && console.log('balancerLensVeBalWeth: ', balancerLensVeBalWeth.address);
+
+  return {
+    veBalDelegatorPCVDeposit,
+    gaugeLensBpt30Fei70WethGauge,
+    balancerLensBpt30Fei70Weth,
+    balancerLensVeBalBal,
+    balancerLensVeBalWeth
+  };
+};
+
+// Do any setup necessary for running the test.
+// This could include setting up Hardhat to impersonate accounts,
+// ensuring contracts have a specific state, etc.
+const setup: SetupUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  // Whitelist our contract for vote-locking on Balancer's governance
+  logging && console.log('Whitelist veBalDelegatorPCVDeposit as a smartwallet on Balancer governance...');
+
+  const BALANCER_AUTHORIZER_ADAPTOR = '0x8F42aDBbA1B16EaAE3BB5754915E0D06059aDd75';
+  await forceEth(BALANCER_AUTHORIZER_ADAPTOR);
+  const balAuthorizerSigner = await getImpersonatedSigner(BALANCER_AUTHORIZER_ADAPTOR);
+
+  // commit a smart wallet checker (the Balancer team will have to deploy a new contract
+  // for this purpose. For this local mock, we add the Angle Protocol's smartwallet checker,
+  // that is already deployed onchain.
+  await contracts.veBal
+    .connect(balAuthorizerSigner)
+    .commit_smart_wallet_checker('0xAa241Ccd398feC742f463c534a610529dCC5888E');
+  await contracts.veBal.connect(balAuthorizerSigner).apply_smart_wallet_checker();
+
+  // add veBalDelegatorPCVDeposit as a smartwallet on the smartwallet checker
+  const ANGLE_MULTISIG_ADDRESS = '0xdC4e6DFe07EFCa50a197DF15D9200883eF4Eb1c8';
+  await forceEth(ANGLE_MULTISIG_ADDRESS);
+  const angleMultisigSigner = await getImpersonatedSigner(ANGLE_MULTISIG_ADDRESS);
+  const abi = ['function approveWallet(address _wallet)'];
+  const smartWalletCheckerInterface = new ethers.utils.Interface(abi);
+  const encodeWhitelistingCall = smartWalletCheckerInterface.encodeFunctionData('approveWallet', [
+    contracts.veBalDelegatorPCVDeposit.address
+  ]);
+  await (
+    await angleMultisigSigner.sendTransaction({
+      data: encodeWhitelistingCall,
+      to: '0xAa241Ccd398feC742f463c534a610529dCC5888E' // SmartWalletChecker
+    })
+  ).wait();
+
```

Comment:
angle?

Reply:
Yeah this `setup()` is a bit f*cked up, I agree. Here's how it works :
- VeToken can define a "smart wallet checker"
- If no "smart wallet checker" is defined, all addresses with bytecode size >0 can't vote-lock or prolong locking on any tokens
- If a "smart wallet checker" is defined, this contract can be used to whitelist contract addresses

The Balancer team doesn't have an on-chain smart wallet checker, their veBAL contract has 0x0000 as the smart wallet checker. So, in the setup, I can't whitelist the new PCVDeposit. I first have to define a smart wallet checker on the veBAL contract. Instead of deploying a mock contract to do this, I'm assigning Angle's smartwallet checker to veBAL, and whitelist us on the Angle smartwallet checker.

Reply:
ah, just a comment would be good